### ISSUE TITLE
Android 6 fatal error.

### ISSUE THREAD

**Issue by GabrieleVolpato** [ryanheise/just_audio#149]
Hello,
I just updated to just_audio 0.3.2 and i get a critical crash with my old Android 6 device (a Xiaomi Redmi 4 pro).
You can reproduce the error with the example project, just by trying to install it.

This is the output of Flutter console:
```
Syncing files to device Redmi 4...
I/ExoPlayerImpl( 9242): Init 4ffa570 [ExoPlayerLib/2.11.4] [markw, Redmi 4, Xiaomi, 23]
E/flutter ( 9242): [ERROR:flutter/shell/platform/android/platform_view_android_jni_impl.cc(43)] java.lang.NoSuchMethodError: No interface method stream()Ljava/util/stream/Stream; in class Ljava/util/List; or its super classes (declaration of 'java.util.List' appears in /system/framework/core-libart.jar)
E/flutter ( 9242): 	at com.ryanheise.just_audio.AudioPlayer.decodeAudioSource(AudioPlayer.java:463)
E/flutter ( 9242): 	at com.ryanheise.just_audio.AudioPlayer.getAudioSource(AudioPlayer.java:434)
E/flutter ( 9242): 	at com.ryanheise.just_audio.AudioPlayer.onMethodCall(AudioPlayer.java:283)
E/flutter ( 9242): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:230)
E/flutter ( 9242): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/flutter ( 9242): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
E/flutter ( 9242): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/flutter ( 9242): 	at android.os.MessageQueue.next(MessageQueue.java:323)
E/flutter ( 9242): 	at android.os.Looper.loop(Looper.java:141)
E/flutter ( 9242): 	at android.app.ActivityThread.main(ActivityThread.java:5601)
E/flutter ( 9242): 	at java.lang.reflect.Method.invoke(Native Method)
E/flutter ( 9242): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:774)
E/flutter ( 9242): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:652)
E/flutter ( 9242): 
F/flutter ( 9242): [FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(942)] Check failed: CheckException(env). 
F/libc    ( 9242): Fatal signal 6 (SIGABRT), code -6 in tid 9242 (t_audio_example)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'Xiaomi/markw/markw:6.0.1/MMB29M/V10.2.2.0.MBEMIXM:user/release-keys'
Revision: '0'
ABI: 'arm64'
pid: 9242, tid: 9242, name: t_audio_example  >>> com.ryanheise.just_audio_example <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
Abort message: '[FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(942)] Check failed: CheckException(env). 
'
    x0   0000000000000000  x1   000000000000241a  x2   0000000000000006  x3   0000000000000000
    x4   0000000000000000  x5   0000000000000001  x6   0000000000000000  x7   0000000000000000
    x8   0000000000000083  x9   0000007fa8a49a70  x10  0000000000000003  x11  0000007fc36ccc78
    x12  0000007fa8a49a70  x13  0000005579654e00  x14  0000005579629740  x15  0000000000000000
    x16  0000007fa8a426a8  x17  0000007fa8a04bdc  x18  0000007fa8dd27e8  x19  0000007fa8e67088
    x20  0000007fa8e66fc8  x21  000000000000000b  x22  0000000000000006  x23  000000000000001b
    x24  0000000000000001  x25  0000005578994018  x26  0000005578f078e0  x27  0000005578f078c8
    x28  0000007fa8da860c  x29  0000007fc36ccbe0  x30  0000007fa8a02378
    sp   0000007fc36ccbe0  pc   0000007fa8a04be4  pstate 0000000020000000
backtrace:
    #00 pc 000000000006abe4  /system/lib64/libc.so (tgkill+8)
    #01 pc 0000000000068374  /system/lib64/libc.so (pthread_kill+68)
    #02 pc 00000000000212f8  /system/lib64/libc.so (raise+28)
    #03 pc 000000000001ba98  /system/lib64/libc.so (abort+60)
    #04 pc 00000000000194fc  /data/app/com.ryanheise.just_audio_example-1/lib/arm64/libflutter.so (offset 0x1240000)
Tombstone written to: /data/tombstones/tombstone_07
Lost connection to device.
```
My Flutter doctor (I don't know why it complains about Android Studio plugin):
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 1.20.1, on Microsoft Windows [Versione 10.0.19041.388], locale it-IT)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[!] Android Studio (version 4.0)
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
[!] VS Code (version 1.47.3)
    X Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[√] Connected device (1 available)

! Doctor found issues in 2 categories.
```

Everything works fine if I try the example project on my Galaxy S8 powered by Android 9.

Kind regards

**Comment by ryanheise** [comment]
I see, thanks for spotting this. I will have to reimplement it without the streams API.

**Comment by ryanheise** [comment]
I have just fixed it on git master. Can you try?

```dart
  just_audio:
    git:
      url: https://github.com/ryanheise/just_audio.git
```

**Comment by GabrieleVolpato** [comment]
I’m going to try it later today, when I get back home from work.

Tank you!

**Comment by GabrieleVolpato** [comment]
I did some tests and everything works great now!

**Comment by ryanheise** [comment]
Thanks for letting me know. I have now published this fix in 0.3.3.
